File: leoz-node/src/main/java/org/deku/leoz/node/messaging/AuthorizationMessageHandler.java
package org.deku.leoz.node.messaging;

import org.deku.leoz.node.messaging.entities.AuthorizationMessage;
import org.jetbrains.annotations.NotNull;
import sx.jms.Converter;
import sx.jms.Handler;

import javax.jms.Message;
import javax.jms.Session;

/**
 * Created by masc on 30.06.15.
 */
public class AuthorizationMessageHandler implements Handler<AuthorizationMessage> {
    @Override
    public void onMessage(AuthorizationMessage message, @NotNull Converter converter, @NotNull Message jmsMessage, @NotNull Session session) {
        // TODO: Push authorization update handling. May revoke the node's authorization key
    }
}

File: leoz-node/src/main/java/org/deku/leoz/node/messaging/AuthorizationMessageHandler.kt
package org.deku.leoz.node.messaging

import org.deku.leoz.node.messaging.entities.AuthorizationMessage
import sx.jms.Converter
import sx.jms.Handler

import javax.jms.ConnectionFactory
import javax.jms.Message
import javax.jms.Session

/**
 * Created by masc on 30.06.15.
 */
class AuthorizationMessageHandler : Handler<AuthorizationMessage> {
    override fun onMessage(message: AuthorizationMessage, converter: Converter, jmsMessage: Message, session: Session, connectionFactory: ConnectionFactory) {
        // TODO: Push authorization update handling. May revoke the node's authorization key
    }
}
